File: Python WorldSimulatorGame/Czlowiek.py
from Organizm import Organizm
from Zwierze import Zwierze
from Punkt import Punkt

class Czlowiek(Zwierze):

    def __init__(self, polozenie, swiat):
        super(Czlowiek, self).__init__(5, 4, 'C', 0, True, polozenie, swiat)
        self.ileOdejmowac= 0

    def getNazwaOrganizmu(self):
        return "Czlowiek"

    def akcja(self):
        self.wiek += 1
        self.ogolneInformacje()
        nowePolozenie = self.swiat.getNastepnyPunktCzlowieka()
        if self.swiat.sprawdzCzyPolePuste(nowePolozenie.getX(), nowePolozenie.getY()):
            self.Ruch(nowePolozenie)
            self.swiat.setPunktCzlowieka(nowePolozenie)
        else:
            ruch = True
            if self.swiat.zwrocOrganizmZPola(nowePolozenie).getZycie():
                self.kolizja(self.swiat.zwrocOrganizmZPola(nowePolozenie), True, ruch)
            if self.getZycie() and ruch:
                org = self.swiat.zwrocOrganizmZPola(nowePolozenie)
                if org is not None and org.getZycie() is False or org is None:
                    self.Ruch(nowePolozenie)
                    self.swiat.setPunktCzlowieka(nowePolozenie)
        if self.getSila() >= 10 and self.swiat.getIloscTurDoSuperMocy() < 5:
            self.swiat.setCzyCzlowiekMozeUzywacSuperMocy(False)
        if self.ileOdejmowac > 0:
            self.setSila(self.getSila() - 1)
            self.ileOdejmowac -= 1

    def kolizja(self, other: Organizm, atakujacy, ruch):
        if self.swiat.getCzySuperMocZostalaAktywowana():
            self.ileOdejmowac = 10 - self.getSila() - (10 - self.swiat.getIloscTurDoSuperMocy())
            if self.ileOdejmowac > 0:
                self.setSila(self.getSila() + self.ileOdejmowac)
            self.swiat.setCzySuperMocZostalaAktywowana(False)
        if other.getZycie():
            if self.getSila() > other.getSila() or (self.getSila() == other.getSila() and atakujacy):
                other.kolizja(self, False, ruch)
            elif self.swiat.getPrzerwij() < 2:
                self.swiat.setPrzerwij(self.swiat.getPrzerwij() + 1)
                other.kolizja(self, False, ruch)
            else:
                self.zabojczeInformacjeDlaZwierzat(other)
                self.setZycie(False)

File: C++ WorldSimulatorGame/POproj1/Czlowiek.cpp
#include "Czlowiek.h"
#include <Windows.h>

Czlowiek::Czlowiek(Punkt polozenie, Swiat* swiat)
	:Zwierze(5, 4, 'C', 0, true, Organizmy::czlowiek, polozenie, swiat) {
	this->sila = 5;
	this->inicjatywa = 4;
	this->znak = 'C';
	this->wiek = 0;
	this->ileOdejmowac = 0;
}
string Czlowiek::getNazwaOrganizmu() {
	return "Czlowiek";
}
void Czlowiek::akcja() {
	this->wiek++;
	//cout << "Punkt 1: to -> " << stary.getX() << " " << stary.getY() << endl;
	ogolneInformacje();
	//cout << this->getNazwaOrganizmu() << " ";
	//cout << " Ma tyle sily: " << this->getSila() << endl;
	//cout << " Z pola " << getPolozenie().getX() << " " << getPolozenie().getY();
	Punkt p = swiat->getNastepnyPunktCzlowieka();
	if (swiat->czyPustePole(p.getX(), p.getY())) {
		//cout << "M1" << endl;
		Move(p);
		swiat->setPunktCzlowieka(p);
		//swiat->Ruch(this, p);
	}
	else {
		//cout << " 1) " << p.getX() << " " << p.getY() << endl;
		//cout << swiat->getMapaSwiata()[p.getY()][p.getX()]->getZnak() << endl;
		bool ruch = true;
		kolizja(swiat->getMapaSwiata()[p.getY()][p.getX()], true, &ruch);
		if (this->getZycie() && swiat->czyPustePole(p.getX(), p.getY()) && ruch) {
			//cout << "M2" << endl;
			Move(p);
			swiat->setPunktCzlowieka(p);
			//swiat->Ruch(this, p);
		}
	}
	if (ileOdejmowac > 0) {
		cout << " odajmowanie sily czlowiekowi " << endl;
		this->setSila(this->getSila() - 1);
		ileOdejmowac--;
	}
	swiat->info.dodajInfocjmacje("\n");
}
void Czlowiek::kolizja(Organizm* other, bool atakujacy, bool* ruch) {

	if (swiat->getCzySuperMocZostalaAktywowana()) {
		this->setSila(this->getSila() + swiat->getIloscTurDoSuperMocy());
		swiat->setCzySuperMocZostalaAktywowana(false);
		ileOdejmowac = swiat->getIloscTurDoSuperMocy();
	}

	if (this->getSila() > other->getSila() || (this->getSila() == other->getSila() && atakujacy)) {
		other->kolizja(this, false, ruch);
	}
	else {
		zabojczeInformacjeDlaZwierzat(other);
		//std::cout << other->getZnak() << " z sila: " << other->getSila() << " zabil -> " << this->getZnak() << " z sila: " << this->getSila() << std::endl;
		swiat->wyczyscOrganizmZMapy(this->polozenie);
		swiat->setZycieCzlowieka(false);
	}
}

void Czlowiek::rysowanie() {
	HANDLE hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
	SetConsoleTextAttribute(hConsole, 6);
	cout << this->znak;
	SetConsoleTextAttribute(hConsole, 7);
}